src/es-module-shims.js

```diff
@@ -64,6 +64,9 @@ async function topLevelLoad (url, source, polyfill) {
 }
 
 async function importShim (id, parentUrl = pageBaseUrl) {
+  console.log('importShim before await', id)
+  await importMapPromise;
```

Comment:
Without this line even the main inline import map isn't yet processed in Safari when the first `importShim` is called.

Reply:
Yes that is what this line is for - to wait on the currently loading import map.

Reply:
Did you verify if adding:

```suggestion
  await Promise.all([importMapPromise, new Promise(resolve => setTimeout(resolve, 0))]);
```

solved the issue here?

Reply:
no, let me try!

Reply:
ah, it's probably not gonna work because the problem is until `new Promise(resolve => setTimeout(resolve, 0))` is resolved `importMapPromise` will be a reference to the "old" promise, that isn't yet chained with the "dynamic import map" promise

Reply:
Makes sense!